### PR TITLE
Ropes refactor flash messages

### PR DESCRIPTION
**Pull request by kkajla12**
Add custom CSS classes extending from Bootstrap card classes specifically for displaying flash messages across the application, and pull flash message markup out into a separate, application-wide partial.

### CODE REVIEW

File: app/assets/stylesheets/_flash-message.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.flash-message {
+  @extend .card;
+}
+
+.flash-message-block {
+  @extend .card-block;
+}
+
+.flash-message-success {
```
Comment (rmacklin) [inline app/assets/stylesheets/_flash-message.scss]: In our BEM notation, this class would be named as a modifier, `flash-message--success`

Relevant article (in particular, the "Name your classes with a logical structure" section - but the whole article is worth reading): http://engineering.appfolio.com/2012/11/16/css-architecture

File: app/views/layouts/_flash_messages.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<% [:error, :success, :notice].each do |flash_type| %>
+  <% if flash[flash_type].present? %>
+    <div class="flash-message flash-message-<%= flash_type %>">
+      <div class="flash-message-block">
+        <p class="white-text"><%= flash[flash_type] %></p>
+      </div>
```
Comment (rmacklin) [inline app/views/layouts/_flash_messages.html.erb]: Seems we can collapse these two elements into one:
```erb
<p class="flash-message-block white-text"><%= flash[flash_type] %></p>
```

File: app/assets/stylesheets/_flash-message.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.flash-message {
+  @extend .card;
+}
+
+.flash-message-block {
```
Comment (rmacklin) [inline app/assets/stylesheets/_flash-message.scss]: This class acts as a subcomponent, so in our BEM notation, that would be `flash-message__block`, although perhaps `flash-message__content` is a more accurate/descriptive name.

File: app/views/layouts/_flash_messages.html.erb (hunk 2 above)
Comment (rmacklin) [inline app/views/layouts/_flash_messages.html.erb]: At that point, you can move the `white-text` styles into your `flash-message-block` class.

File: app/assets/stylesheets/_flash-message.scss (hunk 1 above)
Comment (kkajla12) [inline app/assets/stylesheets/_flash-message.scss]: I wasn't sure whether or not to do this. Now I know. Resolved in 29e09556dc9f97cd46fcb883bfe1c58f6fc4e5d3

File: app/views/layouts/_flash_messages.html.erb (hunk 2 above)
Comment (kkajla12) [inline app/views/layouts/_flash_messages.html.erb]: Good catch. Resolved in 87c9589a4fc37a49c9e0f900ed741f688526b72f and 8a009b7bedc061f660f133b8f8b2f6a9d577015d

File: app/assets/stylesheets/_flash-message.scss (hunk 3 above)
Comment (kkajla12) [inline app/assets/stylesheets/_flash-message.scss]: I knew that! Just missed it. Resolved in 87c9589a4fc37a49c9e0f900ed741f688526b72f

**Comment by rmacklin** [comment]
Looks good. You can go ahead and squash

**Pull request merged**